docs/modules/vector-search/pages/vector-collections.adoc

```diff
@@ -0,0 +1,498 @@
+            </index>
+        </indexes>
+    </vector-collection>
+</hazelcast>
+----
+--
+YAML::
++
+--
+[source,yaml]
+----
+hazelcast:
+  vector-collection:
+    vector-collection-name:
+      indexes:
+        - name: index-1
+          dimension: 2
+          metric: DOT
+        - name: index-2
+          dimension: 2
+          metric: DOT
+          max-degree: 10
+          ef-construction: 20
+          use-deduplication: false
+----
+--
+Java::
++
+--
+[source,java]
+----
+Config config = new Config();
+VectorCollectionConfig collectionConfig = new VectorCollectionConfig("vector-collection-name")
+    .addVectorIndexConfig(
+            new VectorIndexConfig()
+                .setName("index-1")
+                .setDimension(2)
+                .setMetric(Metric.DOT)
+    ).addVectorIndexConfig(
+            new VectorIndexConfig()
+                .setName("index-2")
+                .setDimension(2)
+                .setMetric(Metric.DOT)
+                .setMaxDegree(10)
+                .setEfConstruction(20)
+                .setUseDeduplication(false)
+    );
+config.addVectorCollectionConfig(collectionConfig);
+----
+--
+====
+
+== Manage collection
+
+=== Create collection
+
+To create a vector collection, use the static method `getCollection(HazelcastInstance instance, VectorCollectionConfig collectionConfig)` of the `VectorCollection` class.
+
+Example:
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+VectorCollection vectorCollection = VectorCollection.getCollection(hazelcastInstance, vectorCollectionConfig);
+----
+--
+====
+
+=== Get existing vector collection
+
+To get a vector collection, use the static method `getCollection(HazelcastInstance instance, String collectionName)` of the `VectorCollection` class.
+
+Example:
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+VectorCollection vectorCollection = VectorCollection.getCollection(hazelcastInstance, collectionName);
+----
+--
+====
+
+=== Optimize collection
+
+An optimization operation may be needed in the following cases:
+
+* To permanently delete nodes from the graph that were marked for removal.
+* After adding a significant number of vectors.
+* If some nodes become unreachable and the search results return fewer vectors than expected.
+
+NOTE: The optimization operation is time-consuming process, and no mutable operations are allowed during this process.
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+VectorCollection vectorCollection = VectorCollection.getCollection(hazelcastInstance, collectionName);
+CompletionStage<Void> result = vectorCollection.optimizeAsync("index_name");
+----
+--
+====
+
+== Manage data
+All methods of `VectorCollection` that work with collection data are asynchronous. The result is returned as a `CompletionStage`. A collection interacts with objects in the form of documents (`VectorDocument`). Each document comprises a value and one or more vectors associated with that value.
```

Comment:
I think we do not need to describe what is obvious from Java API (async, CompletionStage) but mention about the role of `VectorDocument` is important

Reply:
but in context of what @EugeneAbramchuk asked in slack, it may be good to mention that concurrency must be controlled when async API is used, so users do not shoot themselves in the foot. BTW, Jet gives that out of the box

Reply:
Do you mean this one? https://hazelcast.slack.com/archives/C06RQRE2MEV/p1717604096835489.
I am not entity understand what you suggest. Do you suggest including information in the documentation that asynchronous operations may incur additional heap overhead?

Reply:
I am not sure what kind heap problem @vbekiaris described, it looks like related to another discussion not to vector search.

with async API it is easy to send many (too many) requests, and each of them will consume heap on server (client also) during processing. but if the client controls the concurrency (relying on clients doing things correctly is of course risky...), then everything should run fine and I am not aware of any other problems specific to async API which do occur also for sync API.

Reply:
> I am not sure what kind heap problem @vbekiaris described, it looks like related to another discussion not to vector search.

right, my note was about JVM heap overheads per `IMap` per partition (that was the topic of the earlier discussion Eugene linked to), unrelated to vector search.

> with async API it is easy to send many (too many) requests, and each of them will consume heap on server (client also) during processing.

That's the issue that's relevant to describe in relation to async requests.

Reply:
Hope you mean to add something like that:
b21e6929889ee3ffe43e883f5baf927a4c76bd29